tests/unit/draw/ElkLayout.spec.js

```diff
@@ -0,0 +1,494 @@
+
+      expect(node).toBeDefined();
+      expect(node?.raw).toBe(component);
+
+      expect(node.parent).toBeDefined();
+
+      expect(node.parent.children[0]).toBe(node);
+
+      expect(node.children).toEqual([]);
+      expect(node.depth).toEqual(1);
+    });
+
+    it('there is a (single) root', () => {
+      const comp1 = new Component({
+        id: 'my_id_1',
+        name: 'my_name_1',
+        drawOption: {},
+        attributes: [],
+        definition: { parentTypes: [] },
+      });
+
+      const comp2 = new Component({
+        id: 'my_id_2',
+        name: 'my_name_2',
+        drawOption: {},
+        attributes: [],
+        definition: { parentTypes: [] },
+      });
+
+      const nodes = layoutManager.getNodes([comp1, comp2]);
+
+      expect(nodes.size).toEqual(2);
+
+      const node1 = nodes.get('my_id_1');
+      const node2 = nodes.get('my_id_2');
+
+      expect(node1?.raw).toBe(comp1);
+      expect(node2?.raw).toBe(comp2);
+
+      expect(node1.parent).not.toBeNull();
+      expect(node1.parent).toBe(node2.parent);
+
+      const root = node1.parent;
+
+      expect(root.children).toEqual([node1, node2]);
+      expect(root.depth).toEqual(0);
+      expect(root.raw).toBeNull();
+      expect(root.parent).toBeNull();
+    });
+
+    it('captures parent-child relationship and depth in the expected way', () => {
+      const comp1 = new Component({
+        id: 'my_id_1',
+        name: 'my_name_1',
+      });
+
+      const comp2 = new Component({
+        id: 'my_id_2',
+        name: 'my_name_2',
+        attributes: [new ComponentAttribute({
+          definition: new ComponentDefinition({ type: 'Reference' }),
+          value: 'my_id_1',
+        })],
+      });
+
+      const comp3 = new Component({
+        id: 'my_id_3',
+        name: 'my_name_3',
+        attributes: [new ComponentAttribute({
+          definition: new ComponentDefinition({ type: 'Reference' }),
+          value: 'my_id_2',
+        })],
+      });
+
+      const nodes = layoutManager.getNodes([comp1, comp2, comp3]);
+
+      const node1 = nodes.get('my_id_1');
+      const node2 = nodes.get('my_id_2');
+      const node3 = nodes.get('my_id_3');
+
+      expect(node1?.raw).toBe(comp1);
+      expect(node2?.raw).toBe(comp2);
+      expect(node3?.raw).toBe(comp3);
+
+      expect(node1.children).toHaveLength(1);
+      expect(node2.children).toHaveLength(1);
+      expect(node3.children).toHaveLength(0);
+
+      expect(node1.children[0]).toBe(node2);
+      expect(node2.children[0]).toBe(node3);
+
+      expect(node2.parent).toBe(node1);
+      expect(node3.parent).toBe(node2);
+
+      expect(node1.parent.depth).toEqual(0);
+      expect(node1.depth).toEqual(1);
+      expect(node2.depth).toEqual(2);
+      expect(node3.depth).toEqual(3);
+    });
+
+    // TODO : error on missing parent / link ? (do we want an error ?)
+  });
+
+  describe('Test method: getNodeDepth', () => {
+    const layoutManager = new ElkLayout();
```

Comment:
for all, declere layout in first line of `it` to be sure that layout is not impacted by other test.

Reply:
done